### DISCUSSION TITLE
[Pytorch] torch class usage question

### DISCUSSION THREAD

**Discussion by logicartis** [bytedeco/javacpp-presets#1128] [Q&A] [answered]
Hello,

A big thank you to the developers for this project. 

I am trying to use the pytorch project. I got stuck on something that is probably really dumb but after about 2 hours, I thought I should ask.

The torch class has methods stack and cat that accept a TensorArrayRef instance. 

In python, one would generally do something like:
```
a = torch.randn([2, 2])
b = torch.randn([2, 2])
torch.cat([a, b], 1)

```
I am trying to replicate the above in Java as below:
```

import static org.bytedeco.pytorch.global.torch.*;
...
Tensor a = randn(2, 2);
Tensor b = randn(2, 2);
// how do I pass both of the above tensors into the cat method?
// TensorArrayRef tensorArray = ???;
// Tensor catted = cat(???, 1);

```
I just cannot figure out how to use the TensorArrayRef class.

Thanks in advance for your help.

**Comment by saudet** [comment, marked as answer]
`c10::ArrayRef<>` is a wrapper around arrays, in this case arrays of `Tensor` objects, so something like this should work:
```java
cat(new TensorArrayRef(new Tensor(2).put(a).position(1).put(b).position(0), 2), 1);
```
But yeah that's not super user-friendly. There is also a constructor taking `std::vector<>`, so I've just mapped that in commit https://github.com/bytedeco/javacpp-presets/commit/eeee3e798af11622fd82686e12d72429444e1c8a. With that we can do something like this instead, which is already much better:
```java
cat(new TensorArrayRef(new TensorVector(a, b)), 1);
```
Please give it a try with the snapshots: http://bytedeco.org/builds/
And thanks for reporting this issue!

**Reply by logicartis** [reply]
Thank you @saudet. Appreciate the help very much.
